### PR TITLE
iOS ScrollView should not scroll out of place on scrolling to element

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change ###

iOS `ScrollView` was behaving differently than Android when using `ScrollToPosition` enums. If you center or end scroll the first element, iOS moves the scrollview out of place by setting the x position to a negative value. Similarly, if you start or center scroll the last element, the x position is set to a value greater than what should be allowed. This change ensures iOS and Android behave the same.

### Bugs Fixed ###

- https://bugzilla.xamarin.com/show_bug.cgi?id=44461

### Behavioral Changes ###

iOS is now more intelligent such that `ScrollView` elements are positioned without going out of bounds. Developers can still set to any position they want by using `ScrollToAsync(double x, double y, bool animated)`.

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs [hunk 1]
```diff
@@ -208,6 +208,25 @@ void OnScrollToRequested(object sender, ScrollToRequestedEventArgs e)
...
+				}
+
+				if (positionOnScroll.Y < 0)
+				{
+					positionOnScroll.Y = 0;
+				}
+				else if (positionOnScroll.Y > ContentSize.Height - Bounds.Size.Height)
+				{
+					positionOnScroll.Y = ContentSize.Height - Bounds.Size.Height;
+				}
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs]: could you use Math.Min() and Math.Max() for this ?

`positionOnScroll.X = Math.Max(positionOnScroll.X, 0);` that'd help the readability

Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs]: well, scratch that, we have a solution for this already:

```
positionOnScroll.X = positionOnScroll.X.Clamp(0, ContentSize.Width - Bounds.Size.Width);
positionOnScroll.Y = positionOnScroll.Y.Clamp(0, ContentSize.Height - Bounds.Size.Height);
```

(Clamp is an internal extension method)

Comment (adrianknight89) [inline Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs]: Done

**Comment by jamesl77** [comment]
Any reason why this and [this](https://github.com/xamarin/Xamarin.Forms/pull/394) were not included in the latest NuGet 2.3.3.163-pre3?

`ScrollView` is a pretty basic control and it's buggy but it looks like someone fixed this, so why not include  the latest NuGet?

**Comment by rmarinho** [comment]
@jamesl77 so 2.3.3 was locked in a stable branch , we only cherry pick things if really really necessary and are related directly with that release and code introduce in that 2.3.3.
This was merged into master and will go in the next version 2.3.4-pre1. 

Thanks

**Comment by jamesl77** [comment]
So you mean this fix will not be included in the 2.3.3 stable release, but in the first 2.3.4 pre-version?
This means another 2 months to wait for such a small but important fix...

**Comment by jamesl77** [comment]
Meanwhile everyone who bumps into this issue will have to bang his head against the wall why on earth `ScrollView`, such a basic control doesn't work properly...

**Pull request merged**